compiler/compile.bal

```diff
@@ -71,9 +71,9 @@ class CompileContext {
 }
 
 // basename is filename without extension
-function compileBalFile(string filename, string basename, string? outputBasename, nback:Options nbackOptions, OutputOptions outOptions) returns CompileError? {
+function compileBalFile(string filename, string basename, string? outputBasename, string? outDir, nback:Options nbackOptions, OutputOptions outOptions) returns CompileError? {
```

Comment:
There's a basic problem here that we haven't yet considered. We need to make this work for files in imported modules. This is handled in processModule/subModuleSourceParts. But for imported modules, it looks like `directory` is being used for the module directory.  We need a design for this.